### PR TITLE
Improve permission check of packages

### PR DESCRIPTION
**Pull request by yp05327**
At first, we have one unified team unit permission which is called `Team.Authorize` in DB.
But since https://github.com/go-gitea/gitea/pull/17811, we allowed different units to have different permission.

The old code is only designed for the old version. So after #17811, if org users have write permission of other units, but have no permission of packages, they can also get write permission of packages.

### CODE REVIEW

**Comment by 6543** [comment]
please add tests to cover that cases :)

**Comment by wolfogre** [comment]
> If org users have write permission of other units, but have no permission of packages, they can also get write permission of packages.

So this is the problem you want to fix, not the thing you want to do, right? It confused me for a minute. 😂

**Review by wolfogre — APPROVED** [review]

**Comment by yp05327** [comment]
I will add tests for this later.

File: models/fixtures/team_unit.yml [hunk 1]
```diff
@@ -268,3 +268,9 @@
   team_id: 9
   type: 1 # code
   access_mode: 1
+
+-
+  id: 46
+  team_id: 17
+  type: 9 # package
+  access_mode: 0
```
Comment (6543) [inline models/fixtures/team_unit.yml]: no newline at the end

**Review by lunny — APPROVED** [review]

File: modules/context/package.go [hunk 2]
```diff
@@ -92,33 +92,25 @@ func determineAccessMode(ctx *Context) (perm.AccessMode, error) {
...
 	accessMode := perm.AccessModeNone
 	if ctx.Package.Owner.IsOrganization() {
 		org := organization.OrgFromUser(ctx.Package.Owner)
 
-		// 1. Get user max authorize level for the org (may be none, if user is not member of the org)
-		if ctx.Doer != nil {
-			var err error
-			accessMode, err = org.GetOrgUserMaxAuthorizeLevel(ctx.Doer.ID)
+		if ctx.Doer != nil && !ctx.Doer.IsGhost() {
+			// 1. If user is logined, check all team packages permissions
```
Comment (delvh) [inline modules/context/package.go]:
```suggestion
			// 1. If user is logged in, check all team packages permissions
```

**Review by delvh — APPROVED** [review]

**Comment by spirkaa** [comment]
This change break access to packages for team members with Administrator Access Permission

**Comment by yp05327** [comment]
> This change break access to packages for team members with Administrator Access Permission. After i changed to General Access with all write perms and change back to Administrator Access, packages stays available. Looks like stale permissions in db or somewhere else.

There is a same problem here: https://github.com/go-gitea/gitea/pull/23675
It will be fixed in #24012

**Pull request merged**